lib/extensions/string.dart

```diff
@@ -119,6 +119,44 @@ extension StringExtensions on String {
   ///Add prefix if not empty else return empty string
   ///
   String get showDashIfEmpty => isEmpty ? '-' : this;
+
+  /// this will give last n characters of string
+  String lastNChars({int n = 1}) {
+    if (n > length || n < 1) return this;
+
+    return substring(length - n);
+  }
+
+  ///this will give first n characters of string
+  String firstNChars({int n = 1}) {
+    if (n > length || n < 1) return this;
+    return substring(0, n);
+  }
+
+  /// Truncates a long `String` in the middle while retaining the beginning and the end.
+  ///
+  /// Example :
+  ///   'Hello World'.truncateMiddle(3) // 'Hel...rld'
+  ///
+  String truncateMiddle({int maxChars = 3}) {
```

Comment:
Add one more parameter for like 
String trucateKey='.'; where it can be defaulted to '.'